### PR TITLE
Avoid infinite loop in vmapped Binomial with p=0

### PR DESCRIPTION
**Pull request by tbenthompson**
Closes https://github.com/pyro-ppl/numpyro/issues/1461

### CODE REVIEW

File: numpyro/distributions/util.py [hunk 1]
```diff
@@ -103,13 +106,19 @@ def accept_fn(k, u, v):
         k, key, u, v = val
         early_accept = (jnp.abs(u) <= tr_params.u_r) & (v <= tr_params.v_r)
         early_reject = (k < 0) | (k > n)
-        return lax.cond(
+        # when vmapped _binomial_dispatch will convert the cond condition into
+        # a HLO select that will execute both branches. This is a workaround
+        # that avoids the resulting infinite loop when p=0. This should also
+        # improve performance in less catastrophic cases.
+        cond_exclude_small_mu = p * n > _binomial_mu_thresh
```
Comment (fehiepsi) [inline numpyro/distributions/util.py:113]: nit: `>` -> `>=`

**Review by fehiepsi — APPROVED** [review]
Woohoo, thanks @tbenthompson! This is tricky to debug and do the right fix.

File: numpyro/distributions/util.py (hunk 1 above)
Comment (tbenthompson) [reply numpyro/distributions/util.py]: good catch!

**Comment by tbenthompson** [comment]
Thanks! I think this is ready to merge any time. Looks like the CI is failing for unrelated reasons to do with FutureWarnings.

**Comment by fehiepsi** [comment]
Thanks, @tbenthompson! Could you help me filter out the warning at [this line](https://github.com/pyro-ppl/numpyro/blob/master/setup.cfg#L22)?

**Comment by martinjankowiak** [comment]
@tbenthompson  you probably want to merge in master after #1464

**Review by fehiepsi — APPROVED** [review]

**Pull request merged**